WooCommerce/Classes/ViewModels/Order Details/Shipping Labels/ReprintShippingLabelViewModel.swift

```diff
@@ -0,0 +1,43 @@
+import Combine
+import Yosemite
+
+/// View model for `ReprintShippingLabelViewController`.
+/// Performs and handles actions that might change data for UI display.
+final class ReprintShippingLabelViewModel {
+    /// Paper size options that we support for reprinting a shipping label.
+    /// In the future, the options could be different per geographical region.
+    let paperSizeOptions: [ShippingLabelPaperSize] = [.legal, .letter, .label]
+
+    /// Observable selected paper size.
+    var selectedPaperSize: AnyPublisher<ShippingLabelPaperSize?, Never> {
+        selectedPaperSizeSubject.eraseToAnyPublisher()
+    }
+
+    private let selectedPaperSizeSubject = CurrentValueSubject<ShippingLabelPaperSize?, Never>(nil)
```

Comment:
I am curious about your approach here. 

My understanding of CurrentValueSubject is that it wraps a single value and published a new element whenever the value changes. 

It seems like the value is expected to change only once? I was wondering if instead of a publisher (selectedPaperSize) publishing a value emitted by a Subject (CurrentValueSubject), it wouldn't be simpler if the view model extended ObservableObject, exposing selectedPaperSize as a read-only property. and the view observed the view model to redraw when the observed object is updated. 

What do you think?

Reply:
The paper size could be changed later by the user in the [paper size selector screen](https://user-images.githubusercontent.com/1945542/101569513-9560c680-3a0f-11eb-8622-46a3a0b1d22d.png) as a separate subtask of https://github.com/woocommerce/woocommerce-ios/issues/2169! In the subtask, there will be a new function added to the view model like `func onPaperSizeSelected(_ paperSize: ShippingLabelPaperSize)`.

Thanks for suggesting `@Published` + `ObservableObject`, I didn't think of using that! It seems like `@Published` is actually implemented by `CurrentValueSubject` underneath but easier to read (just like a property). Since there's no error handling in this use case, I'll update it to `@Published` + `ObservableObject`!

Reply:
Updated in: https://github.com/woocommerce/woocommerce-ios/pull/3341/commits/9a19560b433100de6cfb236856cef2f2f23134a3 - it looks much simpler 😃

Reply:
This is pretty cool! It turns out that you also **don't need** to conform to `ObservableObject`. The `@Published` itself is just fine in my test:

```swift
final class ReprintShippingLabelViewModel {
    /// Observable selected paper size.
    @Published private(set) var selectedPaperSize: ShippingLabelPaperSize?
```

Conforming to `ObservableObject` exposes a `objectWillChange` observable which can be misleading. 

<img src="https://user-images.githubusercontent.com/198826/102102071-a1ba9a00-3de8-11eb-93d9-b08e3f736cb3.png" width="500">

Unless there was something else we were using `ObservableObject` for. 😬

Reply:
@shiki good point! From the [Apple documentation on `@Published`](https://developer.apple.com/documentation/combine/published) and [`ObservableObject`](https://developer.apple.com/documentation/combine/observableobject), it looks like we don't need `ObservableObject` since we are not listening for `objectWillChange`. Updated here: https://github.com/woocommerce/woocommerce-ios/pull/3341/commits/5c5c96a31c1c99e4c8cd43f74b8ecfc21a1571a0